specifyweb/frontend/js_src/lib/components/Header/Logo.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import { commonText } from '../../localization/common';
+import { usePref } from '../UserPreferences/usePref';
+
+export function Logo({
+  isCollapsed,
+  isHorizontal,
+}: {
+  readonly isCollapsed: boolean;
+  readonly isHorizontal: boolean;
+}): JSX.Element {
+  const [logo] = usePref('header', 'appearance', 'customLogo');
+  const [collapsedLogo] = usePref(
+    'header',
+    'appearance',
+    'customLogoCollapsed'
+  );
+
+  return (
+    <h1 className="contents">
+      <a
+        className={`
+      flex items-center gap-2
+      ${isCollapsed ? 'p-2' : 'p-4'}
+      ${isHorizontal ? '' : 'flex-col'}
+    `}
+        href="/specify/"
+      >
+        {/* Both logs are loaded to prevent flickering on collapse/expand */}
+        <img
+          alt=""
+          className={`
+        hover:animate-hue-rotate
+        ${isCollapsed ? 'hidden' : ''}
+      `}
+          src="/static/img/logo.svg"
+        />
+        <img
+          alt=""
+          className={`
+      hover:animate-hue-rotate
+      ${isCollapsed ? '' : 'hidden'}
+      ${isHorizontal ? 'w-10' : ''}
+    `}
+          src="/static/img/short_logo.svg"
+        />
+        {logo === '' ? (
+          ''
+        ) : (
+          <img
+            alt=""
+            className={`max-h-[theme(spacing.24)] max-w-[theme(spacing.24)]
+        hover:animate-hue-rotate
+        ${isCollapsed ? 'hidden' : ''}
+      `}
+            src={logo}
```

Comment:
Please add `crossOrigin="anonymous"` prop to this and the next image.

Read online why it is needed (otherwise browser might block the image from being displayed)